dist/angular-form-messages.js

```diff
@@ -9,49 +9,43 @@ angular.module('angularFormMessages').directive('afField', ["$rootScope", "MESSA
     priority: 100,
     require: ['ngModel', 'afField', '^afSubmit', '^form'],
     controller: function () {
-      function setMessage(type) {
+      function setMessageDetails(type) {
         return function (key) {
-          this.$messages[key] = {
+          ctrl.$messages[key] = {
             type: type
           };
         };
       }
 
+      var ctrl = this;
+
       // Object for storing extra message data such as message type
       this.$messages = {};
 
-      this.setError = setMessage(MESSAGE_TYPES[3]);
-      this.setWarning = setMessage(MESSAGE_TYPES[2]);
-      this.setInfo = setMessage(MESSAGE_TYPES[1]);
-      this.setSuccess = setMessage(MESSAGE_TYPES[0]);
+      this.setMessageDetails = function (key, type) {
+        setMessageDetails(type)(key);
+      };
+      this.setErrorDetails = setMessageDetails(MESSAGE_TYPES[3]);
+      this.setWarningDetails = setMessageDetails(MESSAGE_TYPES[2]);
+      this.setInfoDetails = setMessageDetails(MESSAGE_TYPES[1]);
+      this.setSuccessDetails = setMessageDetails(MESSAGE_TYPES[0]);
     },
     link: function linkFn($scope, elem, attrs, ctrls) {
-      var ngModel = ctrls[0];
-      var afField = ctrls[1];
-      var submit = ctrls[2];
-      var form = ctrls[3];
-
-      function hasValidationChangedAndDirty() {
-        if (ngModel.$dirty && submit.triggerOn === 'change') {
-          updateValidation();
-        }
-      }
-
-      function validationTrigger(newVal, oldVal) {
-        if (oldVal !== newVal) {
-          updateValidation();
-        }
-      }
+      var
+        ngModel = ctrls[0],
+        afField = ctrls[1],
+        submit = ctrls[2],
+        form = ctrls[3],
+        triggerOn = attrs.afTriggerOn || submit.triggerOn || MessageService.triggerOn(),
+        isPristineAfterSubmit;
 
-      /*
-       * Collects validation info from ngModel and afField and passes it to submit.validate()
+      /**
```

Comment:
Hier worden jsdoc comments gebruikt als verduidelijking van een internal functie. Ik zou 1 van 2 dingen kiezen: of volledige jsdoc met parameter en return specificatie; of alleen een line comment.